presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -77,7 +78,8 @@ public JoinNode(
             @JsonProperty("leftHashVariable") Optional<VariableReferenceExpression> leftHashVariable,
             @JsonProperty("rightHashVariable") Optional<VariableReferenceExpression> rightHashVariable,
             @JsonProperty("distributionType") Optional<DistributionType> distributionType,
-            @JsonProperty("dynamicFilters") Map<String, VariableReferenceExpression> dynamicFilters)
+            @JsonProperty("dynamicFilters") Map<String, VariableReferenceExpression> dynamicFilters,
+            @JsonProperty("mergeJoinEnabled") boolean mergeJoinEnabled)
```

Comment:
I don't think mergeJoinEnabled is the right name here.  That sounds like a session property, where as this should be actually setting a different join type.    isMergeJoin would be a better name.